testcode/lib/em-test.c

```diff
@@ -85,9 +117,27 @@ int main (void)
     unsigned PageCount;
     unsigned char X, Y;
     struct em_copy c;
+    unsigned index;
+    signed char valid_key = -1;
+    char key;
+
+    clrscr ();
+    cputs ("Which RAM exp to test?\r\n\r\n");
+    for (index = 0; drivers[index].displayname; ++index) {
+        cprintf("%c: %s\r\n", drivers[index].key, drivers[index].displayname);
+    }
+
+    while (valid_key == -1) {
```

Comment:
For future reference:
C code such as `valid_key < 0` can be optimized by cc65 into better ML code.